### ISSUE TITLE
Problem with downloading videos on FF

### ISSUE THREAD

**Issue by kuzn123** [TheFantasticWarrior/chrome-extension-imagus#2]
0.10.6.4 (FF + Cent)

On Cent all ok. 

Problem with downloading videos on FF DE 112 - by CTRL+S the download does not work (nothing happens). Perhaps it has something to do with the added code to check the rights to download in the browser?

**Comment by TheFantasticWarrior** [comment]
It's actually caused by me trying to solve no file extensions(jpg, mp4 etc.), try this version
[imagus_mod-0.10.6.5.zip](https://github.com/TheFantasticWarrior/chrome-extension-imagus/files/11170290/imagus_mod-0.10.6.5.zip)

**Comment by kuzn123** [comment]
> try this version

I do not know how to add this version to FF.

UPD 
Figured out how to do it, installed, checked - everything seems to be working. I think you can upload this version to the repository for updating.

**Comment by TheFantasticWarrior** [comment]
I'm worried about other sites/files, since I modified the ff download quite a lot, can you test that every sites download(file extensions) is ok before I upload and get another complaint about something broke? Sorry for the late reply didn't get a notification

**Comment by kuzn123** [comment]
0.10.6.5 FF
The saving of images is normal, on those sites that I have tested (so far a little) - extensions are saved (gif, png, jpg). 

But to download some videos on Reddit (HLS/DASH?) CTRL+S combination does not work, example:
https://www.reddit.com/domain/v.redd.it/new/
(there are no problems with Chromium on 0.10.6.4).

There is no way to check everything, I will try to test more during the day.

**Comment by TheFantasticWarrior** [comment]
>0.10.6.5 FF
>The saving of images is normal, on those sites that I have tested (so far a little) - extensions are saved (gif, png, jpg). 

With the new method it saves to jpeg instead jpg when it doesn't come with a default extension, would that cause annoyance to users? Currently I just use whatever comes after the / in [here](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Common_types), probably fixed in [imagus_mod-0.10.6.6.zip](https://github.com/TheFantasticWarrior/chrome-extension-imagus/files/11179920/imagus_mod-0.10.6.6.zip)

>But to download some videos on Reddit (HLS/DASH?) CTRL+S combination does not work, example:
>https://www.reddit.com/domain/v.redd.it/new/
>(there are no problems with Chromium on 0.10.6.4).

I can only open images (thumbnails), not videos, are there other links? Or is this a bug too? it shows old reddit for me and none of the combinations in faq p20 works
edit: v.redd.it works but external-preview.redd.it links don't. seems like a sieve issue
>There is no way to check everything, I will try to test more during the day.

Thank you for helping!

**Comment by kuzn123** [comment]
> probably fixed in [imagus_mod-0.10.6.6.zip]

Unfortunately, nothing has changed on Reddit. Either there is no reaction to pressing CTRL+S, or it offers to save the page (HTML).

> I can only open images (thumbnails), not videos, are there other links? Or is this a bug too? it shows old reddit for me and none of the combinations in faq p20 works

Do you have "autoplay" enabled in your browser? Do other videos work?

Perhaps it's just that you need to "activate the page" - click on it - and only then hover the cursor over the video? FF sometimes shows a "frozen frame" if there is no confirmation of the user's actions on the page.

Setting up sieves for OLD.Reddit:
https://i.imgur.com/Tqk2OL2.png
Examples:
https://OLD.reddit.com/domain/v.redd.it/new/

You should also have enabled:
[HLS_Player-Test]
[Extension]

PS
When importing new sieves, it is very important to completely delete the old ones and click Save. It is even advisable to delete the sieves, press Save, exit the Settings, go back into them and only after that import sieves.

**Comment by kuzn123** [comment]
I checked some of the sites on FF (0.10.6.6). So far I've found this:


FEX.NET
https://fs11.fex.net/preview/3984190050/0x0
0x0.undefined / Cent (0.10.6.4): 0x0.jpg


XUP.IN
https://www0.xup.in/exec/ximg.php?fid=27927804
ximg.php / Cent (0.10.6.4): 2014-04-27_00001.JPG


https://www.flickr.com/photos/chris_rankin/
The first time downloads a video or picture normally, the second time offers to download HTML.


https://www.kinopoisk.ru/film/427076/
(Hover over the poster)
orig.undefined / Cent (0.10.6.4): orig.jpg


https://www.renderotica.com/gallery.aspx?page=4
GetGalleryItem.ashx / Cent (0.10.6.4): GetGalleryItem.jpg

**Comment by TheFantasticWarrior** [comment]
for reddit(url is www but it's the old layout, I also tried old url with old and new sieve combination) it just shows the external-preview.redd.it preview image instead of v.redd.it video. Sometimes after reloading it works but sometimes it doesn't. 
Forgot to mention the times it shows a video download works normally(Dash720.mp4 or dash360.mp4)
.undefined should be fixed [imagus_mod-0.10.6.7.zip](https://github.com/TheFantasticWarrior/chrome-extension-imagus/files/11182523/imagus_mod-0.10.6.7.zip) 

> XUP.IN https://www0.xup.in/exec/ximg.php?fid=27927804 ximg.php / Cent (0.10.6.4): 2014-04-27_00001.JPG

tested on 10.6.0, it should be firefox's behavior itself being different from chrome, or does it work in prior version/original imagus
> https://www.flickr.com/photos/chris_rankin/ The first time downloads a video or picture normally, the second time offers to download HTML.

it works normally for me in 10.6.6(10.6.0 video download has no file extension)
> https://www.renderotica.com/gallery.aspx?page=4 GetGalleryItem.ashx / Cent (0.10.6.4): GetGalleryItem.jpg

yellow spinning circle on every image I tested

**Comment by kuzn123** [comment]
FF DE 112 0.10.6.7


FEX.NET
> .undefined should be fixed imagus_mod-0.10.6.7.zip

Yes, it works normally in this version.


XUP.IN
> tested on 10.6.0, it should be firefox's behavior itself being different from chrome, or does it work in prior version/original imagus

Yes, it's not your mod's problem, it's the behavior of FF itself. The original Imagus can't save correctly either.


FLICKR.COM
https://www.flickr.com/photos/chris_rankin/
Here is the same problem as on version 0.10.6.6 - the first download (by CTRL+S) is normal, the second - HTML. I tried the original Imagus - it also happens there sometimes. It seems that after saving the content, the site "intercepts" the focus on hot keys from Imagus.

At the same time, on the Cent (0.10.6.4) - everything is fine.

I think I found a way to avoid this. You need to hover the cursor, wait for the pop-up to appear (completely), and then do "full zoom" (activation with right mouse button by click). If the pop-up is "activated", then saving occurs normally. But it's not always convenient to "activate" a pop-up for downloading every time, especially when you save a lot of pictures...


RENDEROTICA.COM
> yellow spinning circle on every image I tested

Apparently, the sieve starts working there only for those who are logged in.


I'll try to test other sites a little more today.

**Comment by kuzn123** [comment]
I noticed one feature of FF when working with hotkeys in Imagus (saving to disk) - some sites seem to require **confirmation of user activity on the page**. You need to click on the page at least once and then the contents of the pop-up are saved normally. 

If you just go to the page and start using Imagus right away - after some time or for the second time, the hotkeys may stop working or work incorrectly (they suggest saving HTML), i.e. Imagus seems to lose "focus". This is definitely happening for me on FLICKR.COM and on LOC.GOV for example, here:
https://www.loc.gov/pictures/search/?q=1860-1870&co=gld

This applies primarily to using "S" instead of "CTRL+S".
Although here:
https://www.flickr.com/photos/chris_rankin/
Even when using "CTRL+S", you still need to constantly click on the page so that the Imagus mod does not lose "focus". This is not only in your mod, the Imagus also behaves like this on this page.

Another way to make hotkeys work normally is to **switch the pop-up to ZOOM mode** by right-clicking on the pop-up.

Accordingly, the question is: is it possible to include in the Imagus code an **imitation of a "click"** (pressing the left mouse button on the page) after each download?

PS
I tested another part of the sieves - so far I have not found any serious problems.

**Comment by TheFantasticWarrior** [comment]
firefox 111.0.1 windows 10 
I don't need to click the page to download after the first click to focus on firefox, don't even need to click the page as you can see in the bottom right input display
I'll try to add the function if I find a way though
https://user-images.githubusercontent.com/39485135/230754651-1fb25b89-e32b-4555-a36f-c7989986bf65.mp4
>XUP.IN
>https://www0.xup.in/exec/ximg.php?fid=27927804
>ximg.php / Cent (0.10.6.4): 2014-04-27_00001.JPG

now it downloads as ximg.jpg is that ok? 
edit: added auto "focus" not sure if it works
[imagus_mod-0.10.6.7.zip](https://github.com/TheFantasticWarrior/chrome-extension-imagus/files/11184829/imagus_mod-0.10.6.7.zip)

**Comment by kuzn123** [comment]
> I don't need to click the page to download after the first click to focus on firefox, don't even need to click the page as you can see in the bottom right input display

Try at least 10 times. 
Try to save not only pictures, but also videos. 
Try "S" instead of "CTRL+S".

https://thumbsnap.com/5U7moKWr

But this is rather an exception to the rule, on most sites, downloading content via hotkeys works (it seems to be, since there is no way to test everything in detail).

> now it downloads as ximg.jpg is that ok?

In this version it is downloaded as ximg.jpg, thx. Only the number of this version is the same as the previous one, it's a little confusing.

**Comment by kuzn123** [comment]
I tested the rest of the sites (the largest and most well-known + hosting).

Found only this:


PUU.SH
"Download failed" (by CTRL+S)
https://www.reddit.com/domain/puu.sh/new/
Cent (01064) - downloads normally, but the names of the files are strange:
ss+(2023-04-08+at+12.29.25).jpg
ss+(2023-04-07+at+03.11.44).jpg


PIXIV
"Download failed" (by CTRL+S)
https://www.reddit.com/domain/pixiv.net/new/
https://www.pixiv.net/en/tags/少女/artworks
But Cent (01064) - also doesn't load anything. It looks more like a problem with the sieve, since the original extension can't save anything either by CTRL+S. Only loading via the context menu works on FF DE and Opera. Cent can't save anything even through context menu.


TWITTER - ok, FACEBOOK - can't check because there is no account.
